charts/argo-templates/templates/diff-bq-table.yaml

```diff
@@ -0,0 +1,275 @@
+          - name: JADE_PROJECT
+            value: '{{ "{{inputs.parameters.jade-bq-project}}" }}'
+          - name: JADE_DATASET
+            value: '{{ "{{inputs.parameters.jade-bq-dataset}}" }}'
+          - name: TABLE
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: GCS_PREFIX
+            value: '{{ "gs://{{inputs.parameters.gcs-bucket}}/{{inputs.parameters.input-prefix}}" }}'
+        command: [bash]
+        source: |
+          {{- range .Files.Lines "scripts/diff-bq-table.sh" }}
+          {{ . }}
+          {{- end }}
+
+    - name: query-ids-to-delete
+      inputs:
+        parameters:
+          - name: table-name
+          - name: join-table-name
+          - name: bq-project
+          - name: bq-dataset
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        image: {{ $schemaImage }}
+        env:
+          - name: PROJECT
+            value: '{{ "{{inputs.parameters.bq-project}}" }}'
+          - name: DATASET
+            value: '{{ "{{inputs.parameters.bq-dataset}}" }}'
+          - name: INPUT_TABLE
+            value: '{{ "{{inputs.parameters.join-table-name}}" }}'
+          - name: OUTPUT_PREFIX
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+        command: [bash]
+        source: |
+          {{- range .Files.Lines "scripts/query-non-null-row-ids.sh" }}
+          {{ . }}
+          {{- end }}
+
+    - name: query-rows-to-append
+      inputs:
+        parameters:
+          - name: table-name
+          - name: join-table-name
+          - name: bq-project
+          - name: bq-dataset
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        image: {{ $schemaImage }}
+        env:
+          - name: PROJECT
+            value: '{{ "{{inputs.parameters.bq-project}}" }}'
+          - name: DATASET
+            value: '{{ "{{inputs.parameters.bq-dataset}}" }}'
+          - name: INPUT_TABLE
+            value: '{{ "{{inputs.parameters.join-table-name}}" }}'
+          - name: TABLE
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+        command: [bash]
+        source: |
+          {{- range .Files.Lines "scripts/query-non-null-rows.sh" }}
+          {{ . }}
+          {{- end }}
+
+    - name: get-bq-table-size
+      inputs:
+        parameters:
+          - name: bq-project
+          - name: bq-dataset
+          - name: bq-table
+      {{ include "argo.retry" . | indent 6 }}
+      script:
+        image: {{ $schemaImage }}
+        env:
+          - name: PROJECT
+            value: '{{ "{{inputs.parameters.bq-project}}" }}'
+          - name: DATASET
+            value: '{{ "{{inputs.parameters.bq-dataset}}" }}'
+          - name: TABLE
+            value: '{{ "{{inputs.parameters.bq-table}}" }}'
+        command: [bash]
+        source: |
+          {{- range .Files.Lines "scripts/get-bq-table-size.sh" }}
+          {{ . }}
+          {{- end }}
+
+    - name: extract-bq-table
```

Comment:
I can see value in having these individual steps extracted out as `WorkflowTemplate`s, but I haven't thought through how to deal with inter-template dependencies when we allow for enabling / disabling pieces selectively